src/telliot_feeds/sources/leth_source.py

```diff
@@ -0,0 +1,112 @@
+from dataclasses import dataclass
+from dataclasses import field
+from typing import Any
+from typing import Optional
+
+from telliot_core.apps.telliot_config import TelliotConfig
+
+from telliot_feeds.dtypes.datapoint import OptionalDataPoint
+from telliot_feeds.pricing.price_service import WebPriceService
+from telliot_feeds.pricing.price_source import PriceSource
+from telliot_feeds.sources.price.spot.coinbase import CoinbaseSpotPriceSource
+from telliot_feeds.sources.price.spot.coingecko import CoinGeckoSpotPriceSource
+from telliot_feeds.sources.price.spot.gemini import GeminiSpotPriceSource
+from telliot_feeds.sources.price.spot.kraken import KrakenSpotPriceSource
+from telliot_feeds.sources.price_aggregator import PriceAggregator
+from telliot_feeds.utils.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class LETHSpotPriceService(WebPriceService):
+    """Custom LETH Price Service"""
+
+    def __init__(self, **kwargs: Any) -> None:
+        kwargs["name"] = "Custom LETH Price Service"
+        kwargs["url"] = ""
+        super().__init__(**kwargs)
+        self.cfg = TelliotConfig()
+
+    def get_leth_eth_ratio(self) -> Optional[float]:
+        # get endpoint
+        endpoint = self.cfg.endpoints.find(chain_id=324)
+        if not endpoint:
+            logger.error("Endpoint not found for network 324 to get total pooled tokens")
+            return None
+        ep = endpoint[0]
+        if not ep.connect():
+            logger.error("Unable to connect endpoint for mainnet to get leth_eth_ratio")
+            return None
+        w3 = ep.web3
+
+        # get total pooled tokens from LETH contract
+        total_pooled_tokens_bytes = w3.eth.call(
+            {
+                "to": "0xE7895ed01a1a6AAcF1c2E955aF14E7cf612E7F9d",
+                "data": "0xb6f81c85",
+            }
+        )
+        total_pooled_tokens_decoded = w3.toInt(total_pooled_tokens_bytes)
```

Comment:
should this be in a try/except block in case api returns something it shouldn't (and it can't convert to Int) so that telliot doesn't seize?

Reply:
It's getting the value from the contract on-chain, but you're right it should have some checks. (especially since the contract is upgradable.